src/validation.cpp

```diff
@@ -4989,6 +4989,14 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
                       coins_count - coins_left);
             return false;
         }
+        if (coin.nHeight > base_height ||
+            outpoint.n >= std::numeric_limits<decltype(outpoint.n)>::max() // Avoid integer wrap-around in coinstats.cpp:ApplyHash
```

Comment:
nit: `#include <limits>`

Reply:
The file is already using `std::numeric_limits`, so it seem cleanup that can be done independent of this change. In fact there a quite a few other files which are missing the include, so I am leaving this for another pull:

```
-src/bench/nanobench.h
-src/bitcoin-util.cpp
-src/blockencodings.cpp
-src/blockencodings.h
-src/bloom.cpp
-src/chain.cpp
-src/chainparams.cpp
-src/fs.cpp
-src/init.cpp
-src/leveldb/util/env_windows.cc
-src/miner.cpp
-src/net_processing.cpp
-src/policy/rbf.cpp
-src/primitives/transaction.h
-src/qt/intro.cpp
-src/qt/optionsdialog.cpp
-src/rpc/blockchain.cpp
-src/rpc/mining.cpp
-src/script/interpreter.cpp
-src/support/lockedpool.cpp
-src/test/fuzz/bloom_filter.cpp
-src/test/fuzz/pow.cpp
-src/test/fuzz/rolling_bloom_filter.cpp
-src/test/fuzz/tx_pool.cpp
-src/test/fuzz/util.cpp
-src/test/fuzz/util.h
-src/test/random_tests.cpp
-src/test/rpc_tests.cpp
-src/test/scriptnum_tests.cpp
-src/test/serialize_tests.cpp
-src/test/sighash_tests.cpp
-src/test/skiplist_tests.cpp
-src/test/util_tests.cpp
-src/test/versionbits_tests.cpp
-src/txmempool.cpp
-src/txrequest.cpp
-src/util/system.h
-src/validation.cpp
-src/wallet/interfaces.cpp
-src/wallet/rpcwallet.cpp
-src/wallet/scriptpubkeyman.cpp
-src/wallet/spend.cpp
-src/wallet/wallet.cpp